### PR TITLE
[TypeDeclaration] Add init assign method call support on ReturnTypeFromStrictNewArrayRector

### LINKED ISSUE #8186: Missing return type
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | e.g. v0.11.5 (invoke `vendor/bin/rector --version`)             |

## Minimal PHP Code Causing Issue

https://getrector.com/demo/c994813e-68d1-44eb-9a5b-8e38be0071e7

## Expected Behaviour

I think one of the rules in `\Rector\Set\ValueObject\SetList::TYPE_DECLARATION` should be able to infer a `array` return type

### PR DESCRIPTION
**Pull request by samsonasik**
Closes https://github.com/rectorphp/rector/issues/8186

### CODE REVIEW

**Comment by samsonasik** [comment]
implemented 🎉  /cc @staabm

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

**Comment by staabm** [comment]
Thanks 🙏

File: rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictNewArrayRector/Source/OrderRepository.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictNewArrayRector\Source;
+
+final class OrderRepository
+{
+    public function fetchAllForBuyer(): array
+    {
+        return [];
+    }
+}
```
Comment (TomasVotruba) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictNewArrayRector/Source/OrderRepository.php:12]: Could you add test fixture with docblock array return type, that is skipped? Just to be sure, thanks

Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictNewArrayRector/Source/OrderRepository.php:12]: This is source file under `Source` directory, not fixture file, so it doesn't add docblock.

Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictNewArrayRector/Source/OrderRepository.php:12]: I see, I will add it.

Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictNewArrayRector/Source/OrderRepository.php:12]: @TomasVotruba here the test fixture for skip docblock return:

- https://github.com/rectorphp/rector-src/pull/4933

Comment (TomasVotruba) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictNewArrayRector/Source/OrderRepository.php:12]: Thank you 🙏

**Pull request merged**